label_studio/ml_models/api.py

```diff
@@ -100,6 +99,28 @@ def _is_project_compatible(self, project):
                         if input.get('type', '') == 'Text':
                             return True
         return False
+    def _build_inputs_payload(self, project):
+        parsed_config = project.get_parsed_config()
+        # using this to remove duplicate inputs 
+        input_map = {}
+        for tag in parsed_config:
```

Comment:
Up to you what you think is most clear, but I think we can simplify this logic by reducing into a set, then just mapping back into a dict. Something like:
```python
unique_tags = {
    (("type", tag["type"]), ("value", tag["value"]))  # tuple of tuples, to make it hashable and still simple to map back to a dict
    for _, tag in parsed_config.items() 
    if ("type" in tag) and ("value" in tag)
}
return list(map(dict, unique_tags))
```

Reply:
Ah, I missed one layer of complexity wrt the "inputs" key, I guess it would be more like:
```
input_tags = [tag["inputs"] for tag in parsed_config.values() if "inputs" in tag]
unique_tags = { 
    (("type", tag["type"]), ("value", tag["value"])) # tuple of tuples, to make it hashable and still simple to map back to a dict 
    for tag in input_tags.values() 
    if ("type" in tag) and ("value" in tag) 
} 
return list(map(dict, unique_tags))
```

Reply:
@pakelley - the code you wrote doesnt take into consideration that tag['inputs'] is of type list of dictionaries and not dictionary, so input_tags becomes a list of list of dictionaries, that is the reason I created an input map to collapse all the list into unique values. I would keep the code as is since that is more clear to me

Reply:
lol I don't know if it's quite leetcode cryptic, but I don't mind either way 😁 
I'd prefer if we at least leave a comment here that we're filtering out non-input tags though, since it's pretty easy to miss this in the current implementation (imo)

Reply:
What do you mean, isnt that the whole point, to get input tags? 
`for input in parsed_config[tag].get('inputs',[]):`